src/scripts/main.js

```diff
@@ -1,3 +1,357 @@
 'use strict';
 
-// write code here
+const root = document.querySelector('body');
+const table = document.querySelector('table');
+const tHead = document.querySelector('thead > tr');
+const tBody = document.querySelector('tbody');
+const tdList = root.querySelectorAll('td');
+
+// When I want edit my table cell, my input have own width, and table stretches.
+// Here I set fixed width for my cells,
+// and in css 100% relative width for my input.
+const setWidth = () => {
+  tdList.forEach(td => {
+    td.style.width = `${parseFloat(getComputedStyle(td).width)}px`;
+  });
+};
+
+setWidth();
+
+function addAttributes() {
+  const tBodyList = [...tBody.children];
+
+  for (const tr of tBodyList) {
+    [...tr.cells].forEach((item, index) => {
+      item.dataset.label = tHead.children[index].textContent.toLowerCase();
+    });
+  }
+}
+
+addAttributes();
+
+// sort table;
+const getNumber = num => {
+  return num.replace(/[\D]+/g, '');
+};
+
+const parseTable = () => {
+  return [...tBody.children].map(row => {
+    const person = {};
+
+    [...row.cells].forEach(cell => {
+      person[cell.dataset.label] = cell.textContent;
+    });
+
+    return person;
+  });
+};
+
+const sortData = (data, columnName, sortType) => {
+  const arrEl = getNumber(data[0][columnName]);
+
+  const callBacks = {
+    'ASC': {
+      forString: (a, b) => a[columnName].localeCompare(b[columnName]),
+      forNum: (a, b) => getNumber(a[columnName]) - getNumber(b[columnName]),
+    },
+    'DESC': {
+      forString: (a, b) => b[columnName].localeCompare(a[columnName]),
+      forNum: (a, b) => getNumber(b[columnName]) - getNumber(a[columnName]),
+    },
+  };
+
+  if (arrEl !== '') {
+    return data.sort(callBacks[sortType].forNum);
+  }
+
+  return data.sort(callBacks[sortType].forString);
+};
+
+const replaceTable = (data) => {
+  tBody.innerHTML = data.map(row => `
+    <tr>
+      <td data-label="name">${row.name}</td>
+      <td data-label="position">${row.position}</td>
+      <td data-label="office">${row.office}</td>
+      <td data-label="age">${row.age}</td>
+      <td data-label="salary">${row.salary}</td>
+    </tr>
+    `).join('');
+};
+
+tHead.addEventListener('click', e => {
+  const sortType = e.target.dataset.sort;
+  const sortColumnName = e.target.textContent.toLowerCase();
+  const data = parseTable();
+
+  switch (sortType) {
+    case 'ASC':
+      e.target.dataset.sort = 'DESC';
+      break;
+
+    case 'DESC':
```

Comment:
```suggestion
    case 'DESC':
    default:
```

no need to write it 2 times